frontend/src/pages/home.tsx

```diff
@@ -25,8 +25,12 @@ export default function Home() {
   }, [])
 
   function joinRoom(roomName: string) {
+    const userName = prompt('Enter your name');
+    if (!userName) {
+      return;
+    }
     socket.emit('joinRoom', roomName);
-    navigate('/game');
+    navigate('/game', { state: { roomName, userName } });
```

Comment:
Question: is this picked up in /game route?

Reply:
Not yet, planning to do in later pr